packages/core/src/codewhisperer/views/activeStateController.ts

```diff
@@ -0,0 +1,145 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { Container } from '../service/serviceContainer'
+import { RecommendationHandler } from '../service/recommendationHandler'
+import { cancellableDebounce } from '../../shared/utilities/functionUtils'
+import { LineSelection, LinesChangeEvent } from '../tracker/lineTracker'
+import { isTextEditor } from '../../shared/utilities/editorUtilities'
+import { RecommendationService } from '../service/recommendationService'
+import { subscribeOnce } from '../../shared/utilities/vsCodeUtils'
+
+export class ActiveStateController implements vscode.Disposable {
```

Comment:
needs a docstring. what is the purpose of this, how does it differ from the numerous other "state" related classes in CW?

Reply:
Gotit, this one is responsible for  updating UI specifically on suggestion action event or line change event. So currently we don't have a single source of truth component who is responsible for this. For example, status bar is updated within `InlineCompletionService.ts`, and I lean to have a dedicated component to handle this instead of spreading them every where in the code base.

Reply:
So it's a good idea to move status bar update logic into this component afterward.

Reply:
maybe I should rename this class and make it more self-explanatory it's an UI updater for suggestion action.